### PR TITLE
AJ-1451: More WDS pacts

### PR DESCRIPTION
**Pull request by jladieu**
This adds pacts to cover additional WDS <-> WSM interactions.

It is paired with consumer changes in https://github.com/DataBiosphere/terra-workspace-data-service/pull/414

### CODE REVIEW

File: service/src/test/java/bio/terra/workspace/pact/WdsContractVerificationTest.java [hunk 1]
```diff
@@ -78,44 +118,183 @@ void pactVerificationTestTemplate(PactVerificationContext context) {
...
+        ControlledAzureStorageContainerResource.builder()
+            .storageContainerName(storageContainerName)
+            .common(
+                ControlledResourceFields.builder()
+                    .name(storageContainerName)
+                    .workspaceUuid(workspaceUuid)
+                    .createdByEmail("storage.container.creator@e.mail")
+                    // TODO(for reviewer): are these reasonable defaults for storage
+                    //   container?
+                    .managedBy(ManagedByType.MANAGED_BY_USER)
```
Comment (jladieu) [inline service/src/test/java/bio/terra/workspace/pact/WdsContractVerificationTest.java:245]: Looking for input on this...I tried to guess at what the most relevant values were, and I'm not sure they _really_ matter but if I can make more reasonable assumptions I'd like to.

Comment (davidangb) [reply service/src/test/java/bio/terra/workspace/pact/WdsContractVerificationTest.java:245]: Does this match what's returned by WSM for a real actual resource? If so, seems fine

**Review by davidangb — APPROVED** [review]
one reply in line, otherwise looks good

File: service/src/test/java/bio/terra/workspace/pact/WdsContractVerificationTest.java (hunk 1 above)
Comment (jladieu) [reply service/src/test/java/bio/terra/workspace/pact/WdsContractVerificationTest.java:243]: I was going off of what was in Swagger & what the validation errors were telling me...good call to just look at a prod resource...adjusted (trivially) in a357b1196e7aaa416646eacd0aca579ba4eecf8b

**Pull request merged**